dask/array/tests/test_optimization.py

```diff
@@ -117,13 +116,6 @@ def test_hard_fuse_slice_cases():
     assert optimize_slices(dsk) == {'x': (getarray, 'x', (None, 5))}
 
 
-def test_dont_fuse_different_slices():
```

Comment:
Yeah, I genuinely don't recall the reason myself either.  I suspect that we were inlining slicing operations overly aggressively.

My recommendation was also a sneaky way to push on my other goal, which is to expose keyword arguments to control rechunk :)